app/src/components/widgets/charts/cumulativeTrendChart/cumulativeTrendChart.jsx

```diff
@@ -0,0 +1,242 @@
+    // (not based on URL construction as in other chart examples)
+  };
+
+  onMouseOut = () => {
+    this.chart.revert();
+  };
+
+  onMouseOver = (id) => {
+    this.chart.focus(id);
+  };
+
+  onClick = (id) => {
+    this.chart.toggle(id);
+  };
+
+  getConfig = () => {
+    const { widget, isPreview, container } = this.props;
+
+    if (!widget || !widget.content || !widget.content.result) {
+      return;
+    }
+
+    const { chartDataColumns, dataGroupNames } = generateChartDataParams(widget);
+    const colors = generateChartColors(widget);
+
+    this.dataGroupNames = dataGroupNames;
+    this.height = container.offsetHeight;
+    this.width = container.offsetWidth;
+
+    this.config = {
+      data: {
+        columns: chartDataColumns,
+        type: 'bar',
+        onclick: this.onChartClick,
+        colors,
+      },
+      axis: {
+        x: {
+          show: !isPreview,
+          type: 'category',
+          categories: this.dataGroupNames.map((category) => {
+            const prefix = widget.contentParameters.widgetOptions.prefix;
+            return category.indexOf(prefix) > -1 ? category.split(`${prefix}:`)[1] : category;
+          }),
+          tick: {
+            centered: true,
+            inner: true,
+          },
+        },
+        y: {
+          show: !isPreview,
+        },
+      },
+      grid: {
+        y: {
+          show: !isPreview,
+        },
+      },
+      size: {
+        height: this.height,
+      },
+      interaction: {
+        enabled: !isPreview,
+      },
+      padding: {
+        top: isPreview ? 0 : 85,
+      },
+      legend: {
+        show: false, // we use custom legend
+      },
+      tooltip: {
+        grouped: false,
+        position: this.getPosition,
+        contents: this.renderContents,
+      },
+    };
+
+    this.setState({
+      isConfigReady: true,
+    });
+  };
+
+  getPosition = (d, width, height) => {
+    const rect = this.node.getBoundingClientRect();
+    const top = this.y - rect.top - height;
+    let left = this.x - rect.left - width / 2;
+
+    if (left < 0) {
+      left = 0;
+    } else if (left + width > rect.width) {
+      left = rect.width - width;
+    }
+
+    return {
+      top: top - 8,
+      left,
+    };
+  };
+
+  getCoords = ({ pageX, pageY }) => {
+    this.x = pageX;
+    this.y = pageY;
+  };
+
+  resizeChart = () => {
+    const newHeight = this.props.container.offsetHeight;
+    const newWidth = this.props.container.offsetWidth;
+
+    if (this.height !== newHeight) {
+      this.chart.resize({
+        height: newHeight,
+      });
+      this.height = newHeight;
+    } else if (this.width !== newWidth) {
+      this.chart.flush();
+      this.width = newWidth;
+    }
+  };
+
+  renderContents = (d) => {
+    const index = d[0].index;
+    const groupName = this.dataGroupNames[index];
+    const columns = this.config.data.columns;
+    const itemsData = [];
+    columns.forEach((column) => {
```

Comment:
maybe `map` is better here? `const itemsData = columns.map`

Reply:
fixed